internal/controller/telemetry/collector_controller.go

```diff
@@ -563,6 +568,59 @@ func normalizeStringSlice(inputList []string) []string {
 	return uniqueList
 }
 
+func appendAdditionalVolumesForTenantsFileStorage(otelCommonFields *otelv1beta1.OpenTelemetryCommonFields, tenants []v1alpha1.Tenant) {
+	var volumeMountsForInit []corev1.VolumeMount
+	var chmodCommands []string
+
+	for _, tenant := range tenants {
+		if tenant.Spec.PersistenceConfig.EnableFileStorage {
+			bufferVolumeName := fmt.Sprintf("buffervolume-%s", tenant.Name)
+			mountPath := storage.DetermineFileStorageDirectory(tenant.Spec.PersistenceConfig.Directory)
+			volumeMount := corev1.VolumeMount{
+				Name:      bufferVolumeName,
+				MountPath: mountPath,
+			}
+			volumeMountsForInit = append(volumeMountsForInit, volumeMount)
+			chmodCommands = append(chmodCommands, fmt.Sprintf("chmod -R 777 %s", mountPath))
```

Comment:
I beleive we should use fsGroup in the podSecurityContext + add the same group id to the securityContext (for the main container)

Reply:
Tried this one out, without much luck, it seems that the recommended solution is to use an init-container as for now: https://github.com/open-telemetry/opentelemetry-helm-charts/issues/650